src/Core/Mesh/TriangleMesh.inl

```diff
@@ -36,55 +32,132 @@ inline TriangleMesh& TriangleMesh::operator=( TriangleMesh&& other ) {
-    ///\todo what about other attribs ?"
-    normals().insert( normals().end(), other.normals().cbegin(), other.normals().cend() );
+    const std::size_t facesBefore = m_faces.size();
     m_triangles.insert( m_triangles.end(), other.m_triangles.cbegin(), other.m_triangles.cend() );
-
-    // Offset the vertex indices in the faces
+    m_faces.insert( m_faces.end(), other.m_faces.cbegin(), other.m_faces.cend() );
+    // Offset the vertex indices in the triangles and faces
     for ( uint t = trianglesBefore; t < m_triangles.size(); ++t )
     {
         for ( uint i = 0; i < 3; ++i )
         {
             m_triangles[t][i] += verticesBefore;
         }
     }
+    for ( uint f = facesBefore; f < m_faces.size(); ++f )
+    {
+        for ( uint i = 0; i < m_faces[f].size(); ++i )
+        {
+            m_faces[f][i] += verticesBefore;
+        }
+    }
+
+    // Deal with all attributes the same way (vertices and normals too)
+    for ( auto& attr : other.m_vertexAttribs.attribs() )
+    {
+        if ( attr->isFloat() )
+        {
+            auto h = m_vertexAttribs.findAttrib<float>( attr->getName() );
+            auto& v0 = static_cast<Attrib<float>*>( &( m_vertexAttribs.getAttrib( h ) ) )->data();
+            const auto& v1 = static_cast<Attrib<float>*>( attr )->data();
+            v0.insert( v0.end(), v1.cbegin(), v1.cend() );
+        } else if ( attr->isVec2() )
+        {
+            auto h = m_vertexAttribs.findAttrib<Vector2>( attr->getName() );
+            auto& v0 = static_cast<Attrib<Vector2>*>( &( m_vertexAttribs.getAttrib( h ) ) )->data();
+            const auto& v1 = static_cast<Attrib<Vector2>*>( attr )->data();
+            v0.insert( v0.end(), v1.cbegin(), v1.cend() );
+        } else if ( attr->isVec3() )
+        {
+            auto h = m_vertexAttribs.findAttrib<Vector3>( attr->getName() );
+            auto& v0 = static_cast<Attrib<Vector3>*>( &( m_vertexAttribs.getAttrib( h ) ) )->data();
+            const auto& v1 = static_cast<Attrib<Vector3>*>( attr )->data();
+            v0.insert( v0.end(), v1.cbegin(), v1.cend() );
+        } else if ( attr->isVec4() )
+        {
+            auto h = m_vertexAttribs.findAttrib<Vector4>( attr->getName() );
+            auto& v0 = static_cast<Attrib<Vector4>*>( &( m_vertexAttribs.getAttrib( h ) ) )->data();
+            const auto& v1 = static_cast<Attrib<Vector4>*>( attr )->data();
+            v0.insert( v0.end(), v1.cbegin(), v1.cend() );
+        }
+    }
+
+    return true;
+}
+
+inline void TriangleMesh::clear() {
+    m_vertexAttribs.clear();
+    m_triangles.clear();
+    m_faces.clear();
+    // restore the default attribs (empty though)
+    initDefaultAttribs();
+}
+
+inline void TriangleMesh::clearAttributes() {
+    const auto v = vertices();
+    const auto n = normals();
+    m_vertexAttribs.clear();
+    initDefaultAttribs();
+    vertices() = v;
+    normals() = n;

```

Comment:
Wait, you mean that you **copy** both the normal and vertices arrays twice in a row ?
This looks terrible, what would be the performances on a 10 millions vertices mesh ?
May I suggest using `std::move` ?

Reply:
using std::swap?

Reply:
`std::exchange` seems more adapted, as it moves data in one single direction, while swap... well, swaps in both directions (x to y and y to x).